libs/ts-rest/core/src/lib/client.ts

```diff
@@ -70,14 +88,36 @@ type DataReturnArgsBase<
    * Additional headers to send with the request, merged over baseHeaders,
    *
    * Unset a header by setting it to undefined
+   *
+   * Mandatory headers defined in the client's baseHeaders are optional
    */
-  headers?: Record<string, string>;
+  headers: Prettify<
+    Merge<
+      'baseHeaders' extends keyof TRouterOptions
+        ? PartialByLooseKeys<
+            ZodInputOrType<TRouterOptions['baseHeaders']>,
```

Comment:
Nit: Can we change `baseHeaders` to maybe `defaultHeaders`? To me, `defaultHeaders` reads better

Reply:
Yeah, I agree that the reads better! We already use `baseHeaders` for the client options, so not sure if it'd be optimal if we are inconsistent with our naming here